packages/flutter/lib/src/material/divider.dart

```diff
@@ -96,25 +116,29 @@ class Divider extends StatelessWidget {
   /// )
   /// ```
   /// {@end-tool}
-  static BorderSide createBorderSide(BuildContext context, { Color color, double width = 0.0 }) {
-    assert(width != null);
+  static BorderSide createBorderSide(BuildContext context, { Color color, double width }) {
     return BorderSide(
-      color: color ?? Theme.of(context).dividerColor,
-      width: width,
+      color: color ?? DividerTheme.of(context).color ?? Theme.of(context).dividerColor,
+      width: width ?? DividerTheme.of(context).thickness ?? 0.0,
     );
   }
 
   @override
   Widget build(BuildContext context) {
+    final double height = this.height ?? DividerTheme.of(context).space ?? 16.0;
+    final double thickness = this.thickness ?? DividerTheme.of(context).thickness ?? 0.0;
+    final double indent = this.indent ?? DividerTheme.of(context).indent ?? 0.0;
+    final double endIndent = this.endIndent ?? DividerTheme.of(context).endIndent ?? 0.0;
+
     return SizedBox(
       height: height,
       child: Center(
         child: Container(
-          height: 0.0,
+          height: thickness,
           margin: EdgeInsetsDirectional.only(start: indent, end: endIndent),
           decoration: BoxDecoration(
             border: Border(
-              bottom: createBorderSide(context, color: color),
+              bottom: createBorderSide(context, color: color, width: thickness),
```

Comment:
Could we do this (the decoration) by just specifying `color: color` on the Container?

Reply:
It would work for `Divider` but the `createBorderSide` function is used in other places like `ListTile.divideTile` so it makes sense to keep the color setting in that function.